contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -102,65 +102,168 @@ contract SellerHandlerFacet is SellerBase {
+            lookups.sellerPendingUpdates[seller.id].clerk = _seller.clerk;
         }
 
         // Notify watchers of state change
         emit SellerUpdated(_seller.id, _seller, _authToken, sender);
+        emit SellerUpdateRoleRequested(_seller.id, sellerUpdate, sender);
+    }
+
+    function approveSellerUpdate(uint256 _sellerId) external {
+        // Cache protocol lookups for reference
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+        address sender = msgSender();
+
+        // Get seller pending update
+        SellerRoleUpdate storage _sellerPendingUpdate = lookups.sellerPendingUpdates[_sellerId];
+
+        require(
+            _sellerPendingUpdate.admin != address(0) ||
+                _sellerPendingUpdate.operator != address(0) ||
+                _sellerPendingUpdate.clerk != address(0),
+            NO_PENDING_SELLER_ROLE_UPDATE
+        );
+        preSellerUpdateChecks(_sellerId, _sellerPendingUpdate);
+
+        // Get storage location for seller
+        (, Seller storage seller, ) = fetchSeller(_sellerId);
+
+        // Approve operator address
+        if (_sellerPendingUpdate.operator == sender) {
+            // If operator changed, update the operator and transfer the ownership of NFT voucher
+            if (seller.operator != _sellerPendingUpdate.operator) {
+                seller.operator = _sellerPendingUpdate.operator;
+
+                IBosonVoucher(lookups.cloneAddress[seller.id]).transferOwnership(seller.operator);
+
+                // Update seller id by operator mapping
+                delete lookups.sellerIdByOperator[seller.operator];
+                lookups.sellerIdByOperator[sender] = _sellerId;
+            }
+
+            // Delete pending update operator
+            _sellerPendingUpdate.operator = address(0);
+        }
+
+        // Approva admin address
+        if (_sellerPendingUpdate.admin == sender) {
+            // Update seller admin
+            if (seller.admin != _sellerPendingUpdate.admin) {
+                seller.admin = _sellerPendingUpdate.admin;
+
+                // Update seller id by admin mapping
+                delete lookups.sellerIdByAdmin[seller.admin];
+                lookups.sellerIdByAdmin[sender] = _sellerId;
+            }
+
+            // Delete pending update admin
+            _sellerPendingUpdate.admin = address(0);
+        }
+
+        // Aprove clerk address
+        if (_sellerPendingUpdate.clerk == sender) {
+            // Update seller clerk
+            if (seller.clerk != _sellerPendingUpdate.clerk) {
+                seller.clerk = _sellerPendingUpdate.clerk;
+
+                // Update seller id by clerk mapping
+                delete lookups.sellerIdByClerk[seller.clerk];
+                lookups.sellerIdByClerk[sender] = _sellerId;
+            }
+
+            // Delete pending update clerk
+            _sellerPendingUpdate.clerk = address(0);
+        }
+
```

Comment:
I think if all three updateable addresses are address(0), delete `sellerPendingUpdates[sellerId]` should be called.

Reply:
We are reverting the transaction if all three updateable addresses are 0

```solidity
require(
            _sellerPendingUpdate.admin != address(0) ||
                _sellerPendingUpdate.operator != address(0) ||
                _sellerPendingUpdate.clerk != address(0),
            NO_PENDING_SELLER_ROLE_UPDATE
        );

```

I can remove this require and add your suggestion.
What do you prefer?